src/pkg_tsl2b/word_processors.py

```diff
@@ -98,59 +97,84 @@ def count_words(text: str) -> Dict[str, int]:
-    assert isinstance(dictionary_of_words_and_counts, dict)
+    assert not dictionary is None
+    assert isinstance(dictionary, dict)
 
-    return dictionary_of_words_and_counts
+    return dictionary
 
 
 def parse_arguments():
-    dictionary_of_arguments = {}
-    parser = argparse.ArgumentParser(prog = "Word Processors", description = "This program processes text.")
+    '''
+    Parses command line arguments into a dictionary of argument names and values
+
+    Keyword arguments:
+        none
+
+    Return values:
+        dictionary_of_arguments: Dict[str, int] -- dictionary of argument names and values
+
+    Side effects:
+        none
+
+    Exceptions raised:
+        none
+
+    Restrictions on when this method can be called:
+        first when this script is executed
+    '''
+
+    dictionary = {}
+    parser = argparse.ArgumentParser(
+        prog = "Word Processors",
+        description = "This program processes text."
+    )
     parser.add_argument("name_of_function", help = "name of function")
     parser.add_argument("path_to_file_of_text", help = "path to file of text")
     args = parser.parse_args()
-    name_of_function = args.name_of_function
-    path_to_file_of_text = args.path_to_file_of_text
-    print(f"name of function: {name_of_function}")
-    print(f"path to file of text: {path_to_file_of_text}")
-    dictionary_of_arguments["name_of_function"] = name_of_function
-    dictionary_of_arguments["path_to_file_of_text"] = path_to_file_of_text
-    return dictionary_of_arguments
+    function_name = args.name_of_function
+    path_to_text_file = args.path_to_file_of_text
+    print(f"name of function: {function_name}")
+    print(f"path to file of text: {path_to_text_file}")
+    dictionary["name_of_function"] = function_name
+    dictionary["path_to_file_of_text"] = path_to_text_file
+    return dictionary
 
 if __name__ == "__main__":
     dictionary_of_arguments = parse_arguments()
     name_of_function = dictionary_of_arguments["name_of_function"]
     path_to_file_of_text = dictionary_of_arguments["path_to_file_of_text"]
-    text = None
-    with open(path_to_file_of_text, 'r') as file:
-        text = file.read()
+    TEXT = None
+    with open(path_to_file_of_text, 'r', encoding = "utf-8") as file:
+        TEXT = file.read()
     base_name = os.path.basename(path_to_file_of_text)
     file_name, extension = os.path.splitext(base_name)
     if name_of_function == "clean_text":
-        cleaned_text = clean_text(text)
-        with open(f"{file_name}_Cleaned{extension}", 'w') as file:
-            file.write(cleaned_text)
+        clean_text = clean_text(TEXT)
+        with open(f"{file_name}_Cleaned{extension}", 'w', encoding = "utf-8") as file:
+            file.write(clean_text)
     elif name_of_function == "tokenize":
-        cleaned_text = clean_text(text)
-        list_of_words = tokenize(cleaned_text)
+        clean_text = clean_text(TEXT)
+        list_of_tokenized_words = tokenize(clean_text)
```

Comment:
I like the naming of the variables and functions.  Some argue it's more text, thus more chances for typos, but I think the readability is well worth the effort.

Reply:
@EfrainOlivaresUVA , I feel refreshed reading this. I have experienced tension with bosses and coworkers about naming conventions.